Common/src/main/java/com/almostreliable/unified/utils/JsonCompare.java

```diff
@@ -101,14 +103,104 @@ public static boolean matches(JsonObject first, JsonObject second, Collection<St
+     * value from the original object under a dummy key called "au_sanitized".
+     * <p>
+     * If the element is not a string primitive, the default object is returned.
+     * @param value The value to sanitize
+     * @param defaultValue The default value to return if the element is not a string primitive
+     * @return The sanitized object or the default value
+     */
+    private static JsonElement createSanitizedObjectOrDefault(JsonElement value, JsonElement defaultValue) {
+        if (value instanceof JsonPrimitive primitive && primitive.isString()) {
+            var newObject = new JsonObject();
+            newObject.addProperty("au_sanitized", primitive.getAsString());
+            newObject.addProperty("count", 1);
+            return newObject;
+        }
+        return defaultValue;
+    }
+
+    /**
+     * Used to sanitize root level JSON elements to make them comparable when the count of 1 is implicit.
+     * <p>
+     * This transforms string primitives, JSON objects and JSON arrays to JSON objects where the
+     * count of 1 is explicitly set. The transformation is only applied to a dummy object and not to
+     * the original recipe, so it can be safely used for comparison.
+     * <p>
+     * If the object doesn't support this transformation, the original object is returned.
+     * @param element The element to sanitize
+     * @return The sanitized element or the original element if it can't be sanitized
+     */
+    @SuppressWarnings("ChainOfInstanceofChecks")
+    private static JsonElement sanitize(JsonElement element) {
+        if (element instanceof JsonArray jsonArray) {
+            JsonArray newArray = new JsonArray();
+            for (JsonElement arrayElement : jsonArray) {
+                newArray.add(sanitize(arrayElement));
+            }
+            return newArray;
+        }
+
+        if (element instanceof JsonObject jsonObject) {
+            var keySet = jsonObject.keySet();
+
+            if (keySet.stream().filter(SANITIZE_KEYS::contains).count() != 1) {
+                return element;
+            }
+
+            // if it has a count property, it needs to be 1, otherwise it's implicit 1 as well and needs sanitizing
+            if (keySet.contains("count") && JsonQuery.of(jsonObject, "count").asInt().filter(i -> i == 1).isEmpty()) {
+                return element;
+            }
+
+            var key = keySet.stream().filter(SANITIZE_KEYS::contains).findFirst().orElseThrow();
+            var sanitized = createSanitizedObjectOrDefault(jsonObject.get(key), jsonObject);
+
+            // ensure the object changed (was sanitized) and that we got a JsonObject
+            //noinspection ObjectEquality
+            if (sanitized == jsonObject || !(sanitized instanceof JsonObject sanitizedObject)) {
+                return jsonObject;
+            }
+
+            // add all other properties from the original object
+            for (var entry : jsonObject.entrySet()) {
```

Comment:
The object stored in `sanitized` only has the value passed to `createSanitizedObjectOrDefault` under the `au_sanitizied` key and the count of 1.
In order to successfully compare them after the transformation, the other properties need to be applied too, otherwise, you'd lose information.

I will extract it as a method, call it `merge`, and add some documentation.